pkg/justinstall/registry.go

```diff
@@ -235,16 +239,28 @@ func (e *RegistryEntry) JustInstall(force bool) {
 	e.CreateShims()
 }
 
-func (e *RegistryEntry) installerURL(arch string) string {
+func (e *RegistryEntry) installerURL(arch string) (string, error) {
 	var url string
 
-	if arch == "x86_64" && e.Installer.X86_64 != "" {
-		url = e.Installer.X86_64
+	if arch == "x86_64" {
+		if e.Installer.X86_64 != "" {
+			url = e.Installer.X86_64
+		} else if e.Installer.X86 != "" {
+			url = e.Installer.X86
+		} else {
+			return "", errors.New("No fallback 32-bit download")
```

Comment:
We should define a custom error type (or at lease define a error as a variable) to make it easier to handle errors all in one place.

Reply:
That will come post-Go 1.11 adoption with an overhaul of the codebase. This is just to address some limitations that we had for a long time.